Quantum plots

johannesjmeyer opened the pull request:
I just added support for the figure* environment in twocolumn mode and I think with this the script is pretty much finished. Now it needs some testing on other machines. It may be that the output results differ due to different LaTeX distributions used.

cgranade commented:
One quick suggestion, but would it be possible to deploy your modifications to `rcParams` as an [`*.mplystyle`](https://matplotlib.org/users/style_sheets.html) file, so that your work here can be reused a bit more easily? Thanks!

johannesjmeyer commented:
Surely! I didn't actually know this was possible! Thanks for that suggestion.

Outlook für Android<https://aka.ms/ghei36> herunterladen

________________________________
From: Chris Granade <notifications@github.com>
Sent: Sunday, March 10, 2019 1:12:00 PM
To: quantum-journal/quantum-journal
Cc: johannesjmeyer; Author
Subject: Re: [quantum-journal/quantum-journal] Quantum plots (#73)


One quick suggestion, but would it be possible to deploy your modifications to rcParams as an *.mplystyle<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fmatplotlib.org%2Fusers%2Fstyle_sheets.html&data=02%7C01%7C%7C48846d38ef1d4577360708d6a5519a15%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636878167219618352&sdata=uJjN5%2BBEYCVwRsXBQpUWZyE4S6XvhW2YtbxzFk04vkk%3D&reserved=0> file, so that your work here can be reused a bit more easily? Thanks!

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fquantum-journal%2Fquantum-journal%2Fpull%2F73%23issuecomment-471282497&data=02%7C01%7C%7C48846d38ef1d4577360708d6a5519a15%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636878167219628351&sdata=8UFnuHCGScqSedvh9EoICOe37GHJjcX3%2BqagyYqX2lU%3D&reserved=0>, or mute the thread<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAst7ICrBpsyijbdTEw0tfaotJllI0-Z3ks5vVPaQgaJpZM4blMXC&data=02%7C01%7C%7C48846d38ef1d4577360708d6a5519a15%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636878167219638362&sdata=pKOXvzm4W2laDo4liqc8elTCT6C3k3PsnjpDmvibk9M%3D&reserved=0>.

cgogolin commented:
The notebook is great. Can you please add a paragraph to `quantum-template.tex` that explains briefly how the code in the notebook can be used and that it is not mandatory but just there to assist our authors. I think before the current Section 4 would be the best place for that.

Also it would be great if you could replace the current Figure 1 with a nice figure from the notebook and add a sentence to the figure caption that links to the new section on plots, i.e., something like: "For more information on how to produce plots that fit the general style of the template with python, please refer to Section 5."

johannesjmeyer commented:
I implemented your suggestions and added a plot and some text to the template, which I strongly encourage you to rewrite because I don't beliebe it's really good yet.

I would not make the plotting script public though -- I sent it to the members of the QMIO group and some had trouble running it. Until this has not been remedied we should keep it in the beta stage.

Would you think that it is a good idea to write a blog post about the script (and maybe some general plotting advice) when it is done?

cgogolin commented on quantum-plots.mplstyle:
```diff
@@ -0,0 +1,21 @@
+legend.framealpha: 1.0
+legend.fancybox: False
+
+axes.edgecolor: 555555
```
```suggestion
axes.edgecolor: 555555

```

cgogolin replied:
Without this git complains about a missing newline at the end of the file...

cgogolin commented on quantum-template.tex:
```diff
@@ -171,6 +169,11 @@ \section{Unpublished and accepted works}
+\section{Plots}\label{sec:plots}
+Quantum provides a Jupyter notebook based on the widely used library \texttt{matplotlib} that greatly simplifies the creation of plots that integrate seamlessly with the \texttt{quantumarticle} class. This is intended as a service to the authors and is \textit{not} mandatory. The notebook can be found in the \href{https://github.com/quantum-journal/quantum-journal}{GitHub repository}. The Python  ...
+
+We strongly encourage authors to use a lossless vector format for their submissions, \textit{e.g.}\ Pdf or Pgf. 
```
```suggestion
We strongly encourage authors to use the PDF format for their submissions.
```

cgogolin replied:
We want authors to use pdflatex on the arXiv, because we need the arXiv to produce a pdf.

johannesjmeyer replied:
I was referring to the format of the figures, but I think just staying with PDF is clearer. The quality of PDF is the same as PGF and you can at least look at it without compiling it first.

cgogolin commented on quantum-template.tex:
```diff
@@ -171,6 +169,11 @@ \section{Unpublished and accepted works}
 On the contrary, by giving the \texttt{accepted=YYYY-MM-DD} option, with \texttt{YYYY-MM-DD} the acceptance date, the note ``Accepted in Quantum YYYY-MM-DD, click title to verify'' can be added to the bottom of each page to clearly mark works that have been accepted in Quantum. 
 
+\section{Plots}\label{sec:plots}
+Quantum provides a Jupyter notebook based on the widely used library \texttt{matplotlib} that greatly simplifies the creation of plots that integrate seamlessly with the \texttt{quantumarticle} class. This is intended as a service to the authors and is \textit{not} mandatory. The notebook can be found in the \href{https://github.com/quantum-journal/quantum-journal}{GitHub repository}. The Python  ...
```
```suggestion
Quantum provides a \href{https://jupyter.org/}{Jupyter notebook} based on the widely used \href{https://matplotlib.org/}{matplotlib} library that greatly simplifies the creation of plots that integrate seamlessly with the \texttt{quantumarticle} class. This is intended as a service to the authors, is \textit{not} mandatory, and currently in beta stage. You can \href{https://github.com/quantum-journal/quantum-journal/blob/quantum-plots/quantum-plots.ipynb}{preview} and download the \href{https://raw.githubusercontent.com/quantum-journal/quantum-journal/master/quantum-plots.ipynb}{quantum-plots.ipynb} notebook and accompanying \href{https://raw.githubusercontent.com/quantum-journal/quantum-journal/master/quantum-plots.mplstyle}{quantum-plots.mplstyle} file from the \href{https://github.com/quantum-journal/quantum-journal}{quantumarticle GitHub repository}. You only need to specify the font size and paper format that were passed as options to \texttt{quantumarticle} to get plots with fitting font sizes.
```

cgogolin replied:
How about that? The links should start working once we merge this into master. We should remember to check that this is actually the case...

cgogolin replied:
Ah, and maybe can you please and `(beta)` to the title of the Notebook? Thanks!

johannesjmeyer commented:
I also did the recompile and added (beta) to the title. 

And I'm happy to write the blog post!
